src/dmrg_x.jl

```diff
@@ -1,13 +1,17 @@
+struct DMRGInfo
+  converged::Int
+end
+
 function dmrg_x_solver(PH, t, psi0; kwargs...)
   H = contract(PH, ITensor(1.0))
   D, U = eigen(H; ishermitian=true)
   u = uniqueind(U, H)
   max_overlap, max_ind = findmax(abs, array(psi0 * U))
   U_max = U * dag(onehot(u => max_ind))
-  return U_max, nothing
+  return U_max, DMRGInfo(-1)
 end
 
 function dmrg_x(PH, psi0::MPS; reverse_step=false, kwargs...)
   t = Inf
-  return tdvp(dmrg_x_solver, PH, t, psi0; reverse_step, kwargs...)
+  return tdvp(dmrg_x_solver, PH, t, psi0; reverse_step, kwargs...)[1]
```

Comment:
Here please catch the return values on a previous line like
```
psi, info = tdvp(...)
```
then put `return psi` on the last line.

Reply:
I have synced my branch with main (which has changed due to the contract_mpo_solver), fixed all the tests, and implemented your suggestion.